core/feconf.py

```diff
@@ -22,11 +22,18 @@
 import datetime
 import enum
 import os
+import sys
 
 from core.constants import constants
 
 from typing import Dict, List, Union
-from typing_extensions import TypedDict
+
+# We require typing_extensions which is present in thiry_party so we add it to
```

Comment:
third_party is misspelled.

That said, I am a bit worried about this, we shouldn't have this sort of thing in the main codebase. It can cause problems with deployment. Why is it needed now when it wasn't before?

Reply:
Fixed typo.
This import used to be present in python_utils, it got moved here. python_utils being a very commonly used file, made sure that third_party was already in path, but after it is removed, the CI failed tests showed that, this was the first file which required this import which means before the imports and path manipulations done in `common.py`, this file is executed and we require some library from third_party.

Reply:
Yep, I agree with Sean, I was actually waiting for the reason this is needed in order to give a recommendation on what should we do with this. I think we need to understand what is the purpose of this and then decide where and if to put it anywhere.

Reply:
@vojtechjelinek @seanlip is it ok, if I leave a TODO here so that I can address this in another PR.
I am unable to debug this effectively as this third_party import only breaks with github CI, actions. I cannot reproduce this locally.

Reply:
No, I think we do need to fix this, since it could potentially impact the deployment process. You may need to use GitHub Actions as your "debugger" if you can't repro locally following whatever the setup script being used in the Actions workflow is -- to do that efficiently, you can temporarily get rid of the other workflows perhaps so that only the one that you care about runs.

Reply:
I got it, I am working on it, hoping to get a fix soon.